### PR TITLE
Remove references of deprecated Store IL Opcode

### PR DESCRIPTION
**Pull request by wbh123456**
Remove all references to the IL Opcodes in the `Store` category listed in #2657.

Did not remove the reference of `TR:iustore` in `DebuggingCounters.cpp` as
the opcode is used in a function argument. Will investigate on this
in the future.

Issue: #2657
Signed-off-by: Bohao(Aaron) Wang aaronwang0407@gmail.com

### CODE REVIEW

File: compiler/optimizer/LoopReducer.cpp [hunk 1]
```diff
@@ -357,18 +357,9 @@ TR_Arraycopy::checkArrayStore(TR::Node * storeNode)
 bool
 TR_ByteToCharArraycopy::checkArrayStore(TR::Node * storeNode)
    {
-   if (storeNode->getOpCodeValue() != TR::cstorei)
```
Comment (andrewcraik) [inline compiler/optimizer/LoopReducer.cpp]: This doesn't look right. I would have thought this should change to `!= TR::sstorei` with appropriate updates within the optimization... As it stands the opt will never do anything now...

File: compiler/optimizer/LoopReducer.cpp [hunk 2]
```diff
@@ -1694,7 +1685,7 @@ TR_Arraytranslate::checkLoad(TR::Node * loadNode)
 bool
 TR_Arraytranslate::checkStore(TR::Node * storeNode)
    {
-   if (storeNode->getOpCodeValue() != TR::cstorei && storeNode->getOpCodeValue() != TR::bstorei)
```
Comment (andrewcraik) [inline compiler/optimizer/LoopReducer.cpp]: This also doesn't look right - I think this needs to change to sstorei with appropriate compensation.

File: compiler/optimizer/LoopReducer.cpp [hunk 3]
```diff
@@ -1750,14 +1741,7 @@ TR_Arraytranslate::checkStore(TR::Node * storeNode)
       }
    else
       {
-      if (storeNode->getOpCodeValue() == TR::cstorei)
```
Comment (andrewcraik) [inline compiler/optimizer/LoopReducer.cpp]: `== TR::sstorei` I think is more appropriate

**Comment by andrewcraik** [comment]
We need to make sure we aren't going to break downstream projects which might be generating nodes with this opcode. They need to stop using the opcode before we merge this.

**Comment by Leonardo2718** [comment]
I assume this is something that's still being worked on.

**Comment by wbh123456** [comment]
This project is still in progress. This PR is to be refinished after https://github.com/eclipse/omr/pull/4092 is merged.

File: compiler/il/OMRIL.cpp [hunk 4]
```diff
@@ -1,5 +1,9 @@
 /*******************************************************************************
+<<<<<<< HEAD
  * Copyright (c) 2000, 2020 IBM Corp. and others
+=======
```
Comment (knn-k) [inline compiler/il/OMRIL.cpp:4]: There are extra lines from git rebase operation.

Comment (wbh123456) [reply compiler/il/OMRIL.cpp]: Sorry for the mistake. Fixed in the new push.

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@wbh123456 : can you address @andrewcraik's comments or comment why they are no longer applicable so they can be resolved?

**Comment by wbh123456** [comment]
> @wbh123456 : can you address @andrewcraik's comments or comment why they are no longer applicable so they can be resolved

Sorry about forgetting to address @andrewcraik 's comments there. The `TR:cstore` should be mapped to `TR::sstore` as what he has commented. I have fixed the mistake in the new push.

File: compiler/optimizer/LoopReducer.cpp (hunk 1 above)
Comment (wbh123456) [reply compiler/optimizer/LoopReducer.cpp:360]: The mistake is fixed in bd43e4f

File: compiler/optimizer/LoopReducer.cpp (hunk 2 above)
Comment (wbh123456) [reply compiler/optimizer/LoopReducer.cpp:1697]: The mistake was fixed in bd43e4f

File: compiler/optimizer/LoopReducer.cpp (hunk 3 above)
Comment (wbh123456) [reply compiler/optimizer/LoopReducer.cpp:1753]: The mistake was fixed in bd43e4f

**Comment by wbh123456** [comment]
@0xdaryl Ping.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**